src/read/xcoff/file.rs

```diff
@@ -275,6 +280,31 @@ pub trait FileHeader: Debug + Pod {
             || (!self.is_type_64() && self.f_magic() == xcoff::MAGIC_32)
     }
 
+    /// Read the auxiliary file header.
+    fn aux_header<'data, R: ReadRef<'data>>(
+        &self,
+        data: R,
+        offset: &mut u64,
+    ) -> Result<Option<&'data Self::AuxHeader>> {
+        let aux_header_size = self.f_opthdr();
+        if aux_header_size == 0 {
+            // No auxiliary header is ok.
+            return Ok(None);
+        }
+        if aux_header_size == xcoff::AOUTHSZ_SHORT {
+            // TODO: skip over the aux header with a size of AOUTHSZ_SHORT.
+            *offset += xcoff::AOUTHSZ_SHORT as u64;
+            return Ok(None);
+        }
+        if aux_header_size != mem::size_of::<Self::AuxHeader>() as u16 {
+            return Err(Error("Invalid auxiliary header size"));
+        }
```

Comment:
I think this is too restrictive. I prefer that we return `Ok(None)` if the size is an unrecognized length.
```suggestion
        if aux_header_size != mem::size_of::<Self::AuxHeader>() as u16 {
            *offset += u64::from(aux_header_size);
            return Ok(None);
        }
```